### ISSUE TITLE
[Bug] Library is empty after importing music.

### ISSUE THREAD

**Issue by keystroke3** [Taiko2k/TauonMusicBox#149]
tl;dr imported music does not show up in the music tab.

I deleted some files from the music folder through the app and downloaded some more music. When I launched the app again, the new songs were not visible, so I rescanned the music folder. It scanned and stoped at 6598. I closed the app, and when I reloaded, `I saw scanning tags [so many remaining]`. I waited until it was done. I restarted the app and it was blank. I uninstalled it using `yay`and reinstalled it using  `trizen` , which you mentioned last time. I deleted the `data folder` in `.local/share/TauonMusicBox`  before the re-installation, just in case. When I opened the app, I imported the music. It took some time to do it, but when it was done, it scanned the tags. I still have an empty home screen showing `playlist is empty drag and drop files to import`. When I drag and drop, the files seem to persist after restarting the app. This only works well with one or two folders, but I have a relatively large music library and this will take forever.

Interestingly, when I check the stats section of the settings, I can see that the songs are in the database.

**Comment by Taiko2k** [comment]
> It scanned and stoped at 6598. 

Sounds like that might be part of the issue there. Try run from terminal then import your music, see if we get an error message.

**Comment by keystroke3** [comment]
I have just run it from the terminal, and I am getting a very strange error.  
`PermissionError: [Errno 13] Permission denied: '/home/ted/.cache/yay/gala/pkg'`. When I remove the `gala` directory, It does the same thing for another package. I tried starting the player as admin in terminal, but when I import the  music, it only says `writing database to disk`. I then decided to rename the yay directory to `oldyay` but it was still finding it and still got the same error but with `../oldyay/...` instead. I even tried relocating to my `/home/ted/` but it still found it. I then decided to move it to an external storage device which I detatched and did the import again. This time it worked!. I can now see my music. This has left me with more questions than answers. I actually think Yay was the problem, but I don't understand why. Why was the Tauon so adamant to open the folders?

**Comment by Taiko2k** [comment]
I'm quite at a loss also. There should be no reason it would access those files. Did it give a source file and line number with that error? (Before it said `PermissionError: [Errno 13] Permission denied:`, like this for example: `Traceback (most recent call last):
  File "tauon.py", line 1, in <module>`

Also strange it would have gotten to the 6598th track then stall that first time. Also don't know how it could track that folder after you moved it, unless something with symlinks was going on.

**Comment by keystroke3** [comment]
> Did it give a source file and line number with that error?  
![Screenshot from 2019-07-26 14-57-18](https://user-images.githubusercontent.com/37796862/61957394-dba6b880-afc7-11e9-9b9c-cad16984b8fb.png)
This is what I was getting.
>Also strange it would have gotten to the 6598th track then stall that first time.  
It took a few seconds to start, maybe 2. It then sped through the reset until 6598 and stalled for like 3 seconds then stalled at 6598 and then the terminal showed the error.

**Comment by Taiko2k** [comment]
Oh, so, that error is appearing while importing. That folder must be inside the folder you're attempting to import... You're drag and dropping your home folder into tauon I take it?

**Comment by keystroke3** [comment]
I also noticed that it imports the whole `/home/ted/` directory instead of just `/home/ted/music`. I looked for a way to remedy this, but failed to find one. I was about to create a new issue about this, but then the failed import issue came up. I chose to address that first. I am guessing that it was also trying to look into the `/home/ted/.cache/yay/` directory but all the subdirectories require root access to open them, which Tauon did not have. But it should have been able to show all the imported tracks after scanning them when reopened Tauon without deleting the data directory the first time.

**Comment by Taiko2k** [comment]
Yeah seems like the issue is that its importing the home folder, rather than the music folder. (Though, I should fix that stall on permission error anyway also).

Might be an issue with finding the XDG music directory. Could you tell me the contents of the file `~/.config/user-dirs.dirs`. Also when you run Tauon from terminal, it should print the line `Found XDG-Music: `, what path does it give for that?

**Comment by keystroke3** [comment]
> Could you tell me the contents of the file ~/.config/user-dirs.dirs 

![Screenshot from 2019-07-27 09-13-33](https://user-images.githubusercontent.com/37796862/61990735-eeb09b80-b04e-11e9-8060-abd8b72a3bdc.png)
> Also when you run Tauon from terminal, it should print the line Found XDG-Music: , what path does it give for that?  

![Screenshot from 2019-07-27 09-11-47](https://user-images.githubusercontent.com/37796862/61990744-143da500-b04f-11e9-8288-1a150b7c2b2f.png)

So I guess the problem is with my OS setup? I don't get why it is showing that. I have never changed the music directory location.

**Comment by Taiko2k** [comment]
Well that explains the issue. 

I wonder if that's a common misconfiguration...  Oh well, have to believe what its set to is intended I guess.

**Comment by keystroke3** [comment]
I hope you can make it skip a directory if it cannot access it instead of stalling. Thanks.
